perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -297,6 +298,18 @@ public Member createSpecificMember(PerunSession sess, Vo vo, Candidate candidate
 		return member;
 	}
 
+	public Member createSponsoredAccount(PerunSession sess, Map<String, String> params, String namespace, ExtSource extSource, String extSourcePostfix, User owner, Vo vo) throws InternalErrorException, ExtendMembershipException, AlreadyMemberException, WrongReferenceAttributeValueException, WrongAttributeValueException, GroupOperationsException {
+		String loginNamespaceUri = "urn:perun:user:attribute-def:def:login-namespace:" + namespace;
+		Map<String, String> generatedParams = getPerunBl().getUsersManagerBl().generateAccount(sess, namespace, params);
+		params.putAll(generatedParams);
+		String extSourceLogin = generatedParams.get(loginNamespaceUri) + extSourcePostfix;
+		UserExtSource userExtSource = new UserExtSource(extSource, 2, extSourceLogin);
+		Candidate candidate = new Candidate(userExtSource, params);
```

Comment:
We can't pass all params, if they contain non-URN like keys (e.g. password). It will cause member creation fail, since during the process we try to get the Attribute by URN and then set it with specified value. When not found it throws InternalErrorException (MembersManagerBlImp - line cca 446).